src/include/daos/tests_lib.h

```diff
@@ -31,9 +31,12 @@
 	do {								\
 		if ((rc) == (expected_rc))				\
 			break;						\
-		print_message("assert_rc_equal: %d != %d\n",		\
-			      rc, expected_rc);				\
+		print_message("Failure assert_rc_equal %s:%d "		\
+			      "%s(%d) != %s(%d)\n", __FILE__, __LINE__, \
+			      d_errstr(rc), rc,				\
+			      d_errstr(expected_rc), expected_rc);	\
 		assert_string_equal(d_errstr(rc), d_errstr(expected_rc)); \
+		assert_int_equal(rc, expected_rc);			\
```

Comment:
Just curious why you're not using fail_msg similar to the assert_success macro above? If we've gotten to this point we know we've failed.

Reply:
I haven't thought about it, but after your comment I did an experiment using fail_msg and the cmocka xml results only had a "Failure!" message instead of the != error details. So I guess that the fail_msg function prevents the assert_string_equal and then the assert_int_equal from happening.

Reply:
Yeah, fail_msg fails the test, so it would skip any calls that happen afterward. Sorry if I wasn't clear. I thought the two assert_*_equal calls could be removed, and print_message replaced with fail_msg (with the same message). Your new message includes all the details I think anyone would need to know.

Reply:
fail_msg doesn't put the string in junit and assert_int_equal spits out hexadecimal.   The reason for the macro in the first place was to get more human readable messages into the junit results which is why we use assert_str_equal.  The extra assert can cover cases (that generally shouldn't happen) where you get two DER_UNKNOWN's in the comparison.  

An alternative would be to disallow it entirely by adding a D_CASSERT to prevent expected_rc from being anything other than 0 or -DER_SOMETHING.  But this is a reasonable patch